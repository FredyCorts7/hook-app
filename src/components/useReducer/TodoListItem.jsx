import React from 'react';

const TodoListItem = ({ todo, index, handleDelete, handleToggle }) => {
  return (
    <li className='list-group-item'>
      <div className='row'>
        <p
          className={`${todo.done ? 'complete' : ''} col-8`}
          onClick={() => handleToggle(todo.id)}
        >
          {index + 1}. {todo.title}
        </p>
        <button
          className='btn btn-outline-danger col-4'
          onClick={() => handleDelete(todo.id)}
        >
          Delete
        </button>
      </div>
    </li>
  );
};

export default TodoListItem;
