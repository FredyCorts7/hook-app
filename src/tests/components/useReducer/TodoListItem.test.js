import React from 'react';
import { shallow } from 'enzyme';
import TodoListItem from '../../../components/useReducer/TodoListItem';
import '@testing-library/jest-dom';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Testing to <TodoListItem />', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const index = 0;

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      index={index}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test('should to show succesfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should to call the handleDelete function', () => {
    wrapper.find('button').simulate('click');

    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('should to call the handleToggle function', () => {
    wrapper.find('p').simulate('click');

    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('should to show correctly paragraph content', () => {
    const receivedText = wrapper.find('p').text();

    expect(receivedText).toBe(`${index + 1}. ${demoTodos[0].title}`);
  });

  test('should have the complete class if Todo.done is true', () => {
    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow(<TodoListItem todo={todo} index={0} />);

    expect(wrapper.find('p').hasClass('complete')).toBeTruthy();
  });
});
