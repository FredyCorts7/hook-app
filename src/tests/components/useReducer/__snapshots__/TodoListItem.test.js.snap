// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <TodoListItem /> should to show succesfully 1`] = `
<li
  className="list-group-item"
>
  <div
    className="row"
  >
    <p
      className=" col-8"
      onClick={[Function]}
    >
      1
      . 
      learn react
    </p>
    <button
      className="btn btn-outline-danger col-4"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</li>
`;
